twister: Remove unrecognised sections test

LukaszMrugala opened the pull request:
The section lists that govern whether the ELF section is recognised or not are out of date.

A programming bug has rendered the unrecognised section check unrunnable for years.

Fix was attempted in #71051, but lack of updated section lists made the revival of the unrecognised section check counter-productive and undesired.

Thus, the practically dead code of the unrecognised section check is removed in this PR.

LukaszMrugala committed:
twister: Remove unrecognised sections test

The section lists that govern whether the ELF
section is recognised or not are out of date.

A programming bug has rendered the unrecognised
section check unrunnable for years.

Thus, the practically dead code of the
unrecognised section check is removed
in this commit.

Signed-off-by: Lukasz Mrugala <lukaszx.mrugala@intel.com>

golowanow commented on scripts/pylib/twister/twisterlib/environment.py:
```diff
@@ -483,12 +483,6 @@ def add_parse_arguments(parser = None) -> argparse.ArgumentParser:
         help="Collect and report ROM/RAM section sizes for each test image built.")
 
-    parser.add_argument(
-        "--disable-unrecognized-section-test",
```
Should we label this option as deprecated first, just in case it is actually in use somewhere downstream, and keep logging errors as warnings for the same ?
With removing just at the surface as it is currently proposed, why `size_calc` still tracks 'recognized' sections ?

golowanow replied:
> so the `--disable-[...]` option is not being deprecated, but becoming the default. 
> Anyone using this option will have the program operation proceed as expected.

with this change:
1) twister will fail if this option is used somewhere, right ?
2) no error logs on unrecognized sections anymore by default

https://github.com/zephyrproject-rtos/zephyr/pull/86257/files#diff-6034a9b2a21a304d491695555d9a7196e6638c8a9598686149c92f49a2c5f1b4L642-L647

LukaszMrugala replied:
2. is the default action now - the check hasn't worked for years.
1. I'll add that information to the release notes.

LukaszMrugala committed:
doc: Inform of option removal

We remove the --disable-unrecognized...,
so we should inform the users of that fact.

Signed-off-by: Lukasz Mrugala <lukaszx.mrugala@intel.com>
